filebeat/input/azureeventhub/eph.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
```

Comment:
@hazcod / @andrewkroh : So I'm not entirely sure how these files got here, but it looks like a bunch of the `x-pack` inputs for filebeat were copied over into the non-`x-pack` folder and then the generator was run, resulting in the huge changeset.

We try and keep a strict separation between Elastic licensed code and the Apache-licensed stuff (with the Elastic licensed stuff in the `x-pack` directory), so these files should not be moved or copied around. Here's the commit where this happened: https://github.com/elastic/beats/pull/18116/commits/bb649180be12fbf874f1e4861da6cd8bcc6e0081.

Since this is core to our project structure and our licensing model, you're likely going to need to revert that commit and fix up the rest of your contribution without it before we can accept it.

Reply:
Ah yes, I believe the reason was that those file need to be moved to be able to compile.

Reply:
@hazcod -- so if you're modifying files in the `x-pack` directory you should still be able to compile/test using the `x-pack` entrypoint. The way these binaries work is that the Apache-licensed binary entrypoint is basically the `filebeat/cmd` sub-package. The `x-pack/filebeat/cmd` sub-package wraps the Apache-licensed entrypoint and includes all of the Elastic-licensed modules.

The Elastic-licensed binary we distribute is here: https://www.elastic.co/downloads/beats/filebeat, while the Apache-licensed binary, without the Elastic-licensed code is here: https://www.elastic.co/downloads/beats/filebeat-oss

So if you're trying to build or test changes involving the Elastic-licensed code, you'll just need to work in the `x-pack/filebeat` directory rather than `filebeat` directly. The `make`/`mage` commands for building and testing should be pretty much the same.

Reply:
Seems like i compiled that wrong, I will open up a new PR this weekend.